CHANGELOG.md

```diff
@@ -4,10 +4,19 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## 2.4.0 - 2023-09-14
+
+### Removed
+
+- Dropped support for Python 3.7, which has reached EOL. (Pull #247)
+
+### Added
+
+- Add official support for Python 3.11. (Pull #247)
 
 ### Fixed
 
+- Use faster SHA512-based key hasher instead of password hashers. Reduces server load by making API key validation orders of magnitude faster (10x to 30x according to estimations, network latency aside). (Pull #244)
```

Comment:
```suggestion
- Use faster SHA512-based key hasher instead of password hashers. Reduces server load by making API key validation orders of magnitude faster (10x to 30x according to estimations, network latency aside). Hashed key will be transparently upgraded the first time `.is_valid()` is called. (Pull #244, Pull #251)
```